### PR TITLE
IZE-303-run-end-to-end-integration-tests-during-pr-via-ci-cd

### PR DESCRIPTION
**Pull request by kobrikx**
#### What's new:

 -  Added smoke-tests initial

### CODE REVIEW

File: .github/workflows/smoke_tests.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: "[Ubuntu]Test IZE"
+name: "Smoke tests"
```
Comment (AutomationD) [inline .github/workflows/smoke_tests.yml:1]: I'd make it simple:
```suggestion
name: "Tests"
```

File: .github/workflows/smoke_tests.yml [hunk 2]
```diff
@@ -36,9 +37,64 @@ jobs:
...
-        run: |
-          mkdir -p ~/.aws && echo -e "[${AWS_PROFILE}]\naws_access_key_id = ${AWS_ACCESS_KEY_ID}\naws_secret_access_key = ${AWS_SECRET_ACCESS_KEY}\nregion = ${AWS_REGION}" >> ~/.aws/credentials
-          cat ~/.aws/credentials
-          cd examples/simple-monorepo
-          ${GITHUB_WORKSPACE}/ize --log-level=debug terraform init
+      - name: Test
+        run: go test --timeout 0 --tags="e2e ecs_apps" ./...
+        env:
+          IZE_EXAMPLES_PATH: examples/ecs-apps-monorepo
+  Bastion:
```
Comment (AutomationD) [inline .github/workflows/smoke_tests.yml:44]:
```suggestion
  Tunnel:
```

File: .github/workflows/smoke_tests.yml [hunk 3]
```diff
@@ -36,9 +37,64 @@ jobs:
       - name: Ize build
```
Comment (AutomationD) [inline .github/workflows/smoke_tests.yml:37]: It's all about ize, so let's not abuse the name.
```suggestion
      - name: Build
```

File: .github/workflows/smoke_tests.yml [hunk 4]
```diff
@@ -9,14 +9,15 @@ env:
...
-  push:
-    branches:
-      - main
+  pull_request:
   workflow_dispatch:
 
 jobs:
-  Build:
-    runs-on: ubuntu-latest
+  Ecs_Apps:
```
[the remark this answers is not in the document]
Comment (kobrikx) [reply .github/workflows/smoke_tests.yml:16]: changed to ecs_apps

File: .github/workflows/smoke_tests.yml [hunk 5]
```diff
@@ -36,9 +37,64 @@ jobs:
       - name: Ize build
         run: go build -o ./ize ./cmd
```
[the remark this answers is not in the document]
Comment (kobrikx) [reply .github/workflows/smoke_tests.yml:38]: We have matrix (ubuntu & macos)  - how do you intend to use artifact with matrix ?

File: examples/ecs-apps-monorepo/.ize/env/testnut/ize.toml [hunk 6]
```diff
@@ -1,4 +1,4 @@
-aws_profile = "nutcorp-dev"
+aws_profile = "default"
```
Comment (AutomationD) [inline examples/ecs-apps-monorepo/.ize/env/testnut/ize.toml:1]: Let's not use profile in ize.toml in the examples please

File: .github/workflows/smoke_tests.yml [hunk 7]
```diff
@@ -7,18 +7,110 @@ env:
...
   AWS_PROFILE: default
+  SSH_PUBLIC_KEY: "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQDZqPTy/u82UzDtTY8GUdtGBrQQ1HoyMxZcv/Sc2FeSJTrcLh3i/Mo8pf/MrTO3G7LxJ8syDrptRFIVUheVY7a38UQvay94rJdYtSy6Z/0wLQbnDhJdhzzaPwXrUpBkZO7pcx2HNaC68D/40gTLEHVygJ09VEx8WS7d5QdS8YI6SnXF8ocLF6Nv1WV73aE8Cvz4aUgJJos7pvyX+/WoGzEwpj1aKVPOswRixf18Eg0G6H6WWDfK6zgqEUxsiiP7TB+yXp6VtbDRoS881PK7Cj3+X+lxMBAAuG/PslM6B6koiulCNWyauBZAuay3WR74dYWblYOpD54x1fzrELe2Q4v/1zTi669VqpVfPlH1cnpYiOXaP1aijCA9GFO/dWjxyAAkw2BVeEcbcDaIWhCToMYTHeq2f8Zzi9pxhZyJVO9qg4bXN3EeDGYrbFiWLdWxF9ITKmoSUYG5v8XhXzVTbAyBzDosDiBaa1pMsqFydgc4uZkKAF9govNw1KhRitovz9s= testnut"
 
 on:
-  push:
-    branches:
-      - main
   workflow_dispatch:
-
+# mv testnut $env
```
Comment (AutomationD) [inline .github/workflows/smoke_tests.yml:14]: Let's remove this

File: .github/workflows/smoke_tests.yml [hunk 8]
```diff
@@ -33,12 +125,96 @@ jobs:
...
         run: |
-          mkdir -p ~/.aws && echo -e "[${AWS_PROFILE}]\naws_access_key_id = ${AWS_ACCESS_KEY_ID}\naws_secret_access_key = ${AWS_SECRET_ACCESS_KEY}\nregion = ${AWS_REGION}" >> ~/.aws/credentials
-          cat ~/.aws/credentials
-          cd examples/simple-monorepo
-          ${GITHUB_WORKSPACE}/ize --log-level=debug terraform init
+          go test --timeout 0 --tags="e2e terraform" ./...
+        env:
+          IZE_EXAMPLES_PATH: ${{ github.workspace }}/examples/ecs-apps-monorepo
+          IZE_BINARY: ${{ github.workspace }}/bin/ize
+          AWS_PROFILE: test-terraform
```
Comment (AutomationD) [inline .github/workflows/smoke_tests.yml:219]: Any reason not to used default here?

File: .github/workflows/smoke_tests.yml [hunk 9]
```diff
@@ -33,12 +125,96 @@ jobs:
...
+          known_hosts: unnecessary
+          if_key_exists: replace
+
+      - name: Run Tests
+        run: |
+          go test --timeout 0 --tags="e2e bastion_tunnel" ./...
+        env:
+          IZE_EXAMPLES_PATH: ${{ github.workspace }}/examples/ecs-apps-monorepo
+          IZE_BINARY: ${{ github.workspace }}/bin/ize
+          AWS_PROFILE: ecs-apps-monorepo
```
Comment (AutomationD) [inline .github/workflows/smoke_tests.yml:159]: Any reason not to use `default` here?

File: .github/workflows/smoke_tests.yml [hunk 10]
```diff
@@ -7,18 +7,110 @@ env:
...
+          known_hosts: unnecessary
+          if_key_exists: replace
+
+      - name: Run Tests
+        run: |
+          go test --timeout 0 --tags="e2e ecs_apps" ./...
+        env:
+          IZE_EXAMPLES_PATH: ${{ github.workspace }}/examples/${{ github.job }}
+          IZE_BINARY: ${{ github.workspace }}/bin/ize
+          AWS_PROFILE: test-ecs-apps
```
Comment (AutomationD) [inline .github/workflows/smoke_tests.yml:98]: any reason not to use `default` here?

File: .github/workflows/smoke_tests.yml [hunk 11]
```diff
@@ -7,18 +7,110 @@ env:
...
+  ecs-apps-monorepo:
+    name: ECS Apps
+    needs: Build
+    strategy:
+      matrix:
+        os: [ ubuntu-latest ]
+    runs-on: ${{ matrix.os }}
+    env:
+      ENV: test-ecs-apps
+      AWS_PROFILE: test-ecs-apps
```
Comment (AutomationD) [inline .github/workflows/smoke_tests.yml:50]: `default`?

**Review by AutomationD — CHANGES_REQUESTED** [review]

**Review by psihachina — APPROVED** [review]

**Review by AutomationD — APPROVED** [review]
Looks good

**Pull request merged**